gensim/matutils.py

```diff
@@ -319,30 +319,48 @@ def veclen(vec):
     assert length > 0.0, "sparse documents must not contain any explicit zero entries"
     return length
 
+def ret_normalized_vec(vec, length):
+    if length != 1.0:
+        return [(termid, val / length) for termid, val in vec]
+    else:
+        return list(vec)
+
+def isnorm(norm):
+    if norm not in ('l1', 'l2'):
+        raise ValueError("'%s' is not a supported norm" % norm)
```

Comment:
I had written it here to avoid code duplication in other modules which need to validate the input norm from user.